app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala

```diff
@@ -299,25 +323,45 @@ object FormTemplateValidator {
   }
 
   def validateChoiceDividerPositionLowerBound(sectionsList: List[Page]): ValidationResult = {
-    def check(choice: Choice): Boolean = choice.dividerPositon.fold(false)(_ <= 0)
+    def check(choice: Choice): Boolean = choice.dividerPosition.fold(false)(_ <= 0)
 
     validateChoice(sectionsList, check, "dividerPosition should be greater than 0")
   }
 
   def validateChoiceDividerPositionUpperBound(sectionsList: List[Page]): ValidationResult = {
-    def check(choice: Choice): Boolean = choice.dividerPositon.fold(false)(_ >= choice.options.size)
+    def check(choice: Choice): Boolean = choice.dividerPosition.fold(false)(_ >= choice.options.size)
 
     validateChoice(sectionsList, check, "dividerPosition should be less than the number of choices")
   }
 
+  def validateChoiceNoneChoiceValue(sectionsList: List[Page]): ValidationResult = {
+    def check(choice: Choice): Boolean = {
+      val values = choice.options.collect { case OptionData.ValueBased(_, v) =>
+        v
+      }
+      choice.noneChoice.fold(false) {
+        case NoneChoice.IndexBased(index) => !values.isEmpty
```

Comment:
`index` is not used anywhere. May be put `_` instead

Reply:
also, isn't this case is a check that noneChoice is index based but there are some ValueBased options. But the error will be `"'noneChoice' should match one of choices 'value'"` which is slightly confusing